### PR TITLE
Remove op validation when inserting into Circuit

### PR DESCRIPTION
**Pull request by dabacon**
Proposal to remove validation on op insertion methods #4667

### CODE REVIEW

File: cirq-core/cirq/ops/raw_types.py
Comment (maffoo) [inline cirq-core/cirq/ops/raw_types.py:854]: I have a slight preference to revert to the previous implementation here because the `qid_shape` protocol seems to be slow, and we know here that it's the same as `tuple(q.dimension for q in qubits)` but we don't actually need to construct the tuple.

File: cirq-core/cirq/ops/raw_types_test.py
Comment (maffoo) [inline cirq-core/cirq/ops/raw_types_test.py:123]: Why this change?

**Review by maffoo — COMMENTED** [review]
I'm in favor of removing this validation from `Circuit`, but would be good to see if other maintainers have objectsions. Just a couple of minor comments.

**Comment by mpharrigan** [comment]
cc @MichaelBroughton re: devices api, which concerns itself with validation

File: cirq-core/cirq/ops/raw_types.py
Comment (dabacon) [reply cirq-core/cirq/ops/raw_types.py]: reverted

File: cirq-core/cirq/ops/raw_types_test.py
Comment (dabacon) [reply cirq-core/cirq/ops/raw_types_test.py]: late night, fixed

**Comment by dabacon** [comment]
Updated per @maffoo's comments.

**Comment by MichaelBroughton** [comment]
Would this change mean a user gets no validation at all on a custom gate they write when putting it into a circuit ?

**Review by maffoo — APPROVED** [review]
LGTM, though may want to get comments from @MichaelBroughton or other maintainers before merging.

**Comment by MichaelBroughton** [comment]
We talked about this in the Cirq meeting today and I think we are fine to merge this in.

**Pull request merged**
